addons/sale/models/sale.py

```diff
@@ -175,7 +175,7 @@ def _get_payment_type(self):
 
     payment_term_id = fields.Many2one('account.payment.term', string='Payment Terms', oldname='payment_term', help="Payment terms applies to all invoices issued from this order.")
     fiscal_position_id = fields.Many2one('account.fiscal.position', oldname='fiscal_position', string='Fiscal Position', help=" Fiscal positions are used to adapt taxes and accounts for particular customers or sales orders/invoices. The default value comes from the customer.")
-    company_id = fields.Many2one('res.company', 'Company', default=lambda self: self.env.company)
+    company_id = fields.Many2one('res.company', 'Company', default=lambda self: self.env.company, required=True)
```

Comment:
As we are making this required we can remove this from master (If they will forwarded into master)
Ref commits
https://github.com/odoo/odoo/pull/34408 https://github.com/odoo/odoo/pull/34374 https://github.com/odoo/odoo/pull/34146 https://github.com/odoo/odoo/pull/34120

(surprisingly all these were reported on span of two weeks)
Cc: @nim-odoo